### DISCUSSION TITLE
How to get AdditionalProperties from Get-PnPAzureADUser

### DISCUSSION THREAD

**Discussion by EA12** [pnp/powershell#1666] [Q&A] [answered]
Hi Experts,

I try to get specific custom attributes from a user object.
I included the field into the -Select paramter

`$users = Get-PnPAzureADUser -Select "Id,mail,UserPrincipalName,companyName,extension_b8fc35d8e8ec45e689d332303177957a_ZZZ_townCode" | Where-Object { $_.AdditionalProperties["extension_b8fc35d8e8ec45e689d332303177957a_ZZZ_townCode"] -ne $null } `

From the keys object, I can find my attributes `$user.AdditionalProperties.Keys`
But I cannot get them through `$user.AdditionalProperties["companyName"]`
or `$user.AdditionalProperties."companyName"`
or `$user.AdditionalProperties.companyName`

The [spec](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/get-pnpazureaduser?view=sharepoint-ps) has no information about it.

What is the recommend way to get the values from my custom attributes?

Regards, Ronny

**Comment by EA12** [comment]
Nobody?

**Comment by AndersRask** [comment]
I have only used extension properties for Get-AzureADUser, and there it worked with

ExtensionProperty.extension_9d98asdfl15980a_Nickname

You could try using that or Get-AzADUser instead, or to test if the property is really there.

**Comment by EA12** [comment, marked as answer]
My bad,
the field name Ii was using was simply wrong. 
I needed extension_b8fc35d8e8ec45e689d332303177957a_company and not companyName.

**Reply by nicole-ge** [reply]
Hi @EA12 ,
how you found the correct name? I need the correct name of the property country and usertype

**Reply by EA12** [reply]
Hi @nicole-ge: the guid is the id of your tenant. Ask your Azure or sharePoint-admin to give you this id. An Azure-admin could also provide the names you want from the Azure AD.

**Comment by nicole-ge** [comment]
Hi @EA12 ,

$users = Get-PnPAzureADUser -Select -Select "Country", "UserType"

Write-Output $azureADUser.AdditionalProperties.country

and it doesn't work? can you please help

**Reply by EA12** [reply]
It should be Get-PnPAzureADUser -Select "country", "userType"
